Fix user deletion with archived meetings

jsangmeister opened the pull request:
fixes #1792 

Changed the user scope calculation to also take archived meetings into account, since meeting admins should not be allowed to delete users which are in their meeting plus in other archived meetings.

r-peschke reviewed (changes requested):
- I'm not convinced that we should change the user scope in this case, but that's to discuss.
- Please fix the checks
- if we do it this way, change the doc in wiki

jsangmeister commented:
Updated the wiki: https://github.com/OpenSlides/OpenSlides/wiki/Users did not mention archived meetings anyway and https://github.com/OpenSlides/OpenSlides/wiki/Archived-Meetings had one hint, which I removed. Also fixed the tests.

jsangmeister commented:
@r-peschke if you update the priority of the PR, you also have to update the priority of the issue. The `critical` evaluation came from @MSoeb, I just copied it from the issue

r-peschke commented:
Wrote comment into the issue as answer to the problem description.

jsangmeister commented:
Reverted the wiki changes. Will update this PR with the changes discussed in https://github.com/OpenSlides/openslides-backend/issues/1795 and https://github.com/OpenSlides/openslides-backend/issues/1792

jsangmeister commented:
Updated the PR with the presenter changes. Also finally implemented the improvement suggested in https://github.com/OpenSlides/openslides-backend/issues/1213 to reduce the amount of filter requests (3 instead of 3 times the amount of meetings for each user).

r-peschke commented on tests/system/presenter/test_get_user_scope.py:
```diff
@@ -11,22 +11,22 @@ def test_good(self) -> None:
+                # archived meeting
+                "meeting/2": {
+                    "committee_id": 2,
+                    "is_active_in_organization_id": None,
```
To be correct `is_archived_in_organization_id` is necessary to determine an archived meeting

r-peschke commented on tests/system/presenter/test_get_user_related_models.py:
```diff
@@ -173,24 +183,72 @@ def test_get_user_related_models_no_permissions(self) -> None:
+                    }
+                ]
+            },
+        }
```
The result still misses the `organization_management_level: OML-String,` info, which is mentioned in the presenter doc. The existing test **test_get_user_related_models_permissions** , beginning at line 279, should have such a result entry, but it is missed and not checked.

jsangmeister replied:
Somehow I thought that the presenter already returned the OML. Added it now

r-peschke reviewed (approved):
Just some smaller changes

r-peschke reviewed (changes requested):
See 3 small changes in the comments

jsangmeister commented:
Added the OML, fixed the tests & comments and reduced the amount of datastore requests in the presenter. Review again @r-peschke

jsangmeister commented:
This still needs changes in the client as described in https://github.com/OpenSlides/openslides-backend/issues/1792#issuecomment-1623203037
